### PR TITLE
Add member variable for base sample rate and inline sample rate getter functions

### PR DESCRIPTION
**Pull request by Rossmaxx**
Optimised calls to `outputSampleRate` by inlining functions and using member variable to speed up access. Makes function calls a lot lesser as this function seems to get called nearly a million times, as a nested call for `outputSampleRate`. 

also inlined the two functions to improve performance

### CODE REVIEW

**Review by sakertooth — COMMENTED** [review]
We might as well also inline the `outputSampleRate` and `inputSampleRate` functions.

**Comment by sakertooth** [comment]
@Rossmaxx, you think you could also inline the functions `outputSampleRate` and `inputSampleRate`? Since these functions are being called in audio code and are just simple getter functions that contain a branch, it seems like the next reasonable thing to do here.

File: include/AudioEngine.h [hunk 1]
```diff
@@ -235,9 +235,18 @@ class LMMS_EXPORT AudioEngine : public QObject
...
+
+	sample_rate_t outputSampleRate() const{
+		return m_audioDev != nullptr ? m_audioDev->sampleRate() : m_baseSampleRate;
+	}
+	
+
+	sample_rate_t inputSampleRate() const{
+		return m_audioDev != nullptr ? m_audioDev->sampleRate() : m_baseSampleRate;
+	}
+
```
Comment (sakertooth) [inline include/AudioEngine.h:249]:
```suggestion
	sample_rate_t outputSampleRate() const { return m_audioDev != nullptr ? m_audioDev->sampleRate() : m_baseSampleRate; }
	sample_rate_t inputSampleRate() const { return m_audioDev != nullptr ? m_audioDev->sampleRate() : m_baseSampleRate; }
```
Just some formatting changes.

**Review by sakertooth — APPROVED** [review]

**Comment by sakertooth** [comment]
Seems like we need to include `AudioDevice.h` within `AudioEngine.h` to fix the build.

File: include/AudioEngine.h (hunk 1 above)
Comment (Rossmaxx) [reply include/AudioEngine.h:249]: Do we really need that? It looks very long.

Comment (sakertooth) [reply include/AudioEngine.h:249]: Fair, maybe just put the braces on there own line? If not it's okay I wont nitpick, its just that the formatting looked strange to me a bit.

**Comment by Rossmaxx** [comment]
Sorry i forgot about this. But can get back after a few days, please wait till then.

**Review by sakertooth — APPROVED** [review]

**Pull request merged**